prune.go

```diff
@@ -27,19 +27,36 @@ const (
 	// logInterval defines often to report progress during long
 	// operations.
 	logInterval = 15 * time.Second
+
+	// maxIterCount is the number of times we will pass over the data to make sure all documents are cleaned up.
+	// (removing from a collection you are iterating can cause you to miss entries). The loop should exit early if
+	// it finds nothing to do anyway, so this only affects the number of times we will evaluate documents we aren't
+	// removing.
+	maxIterCount = 5
+
+	// maxMemoryTokens caps our in-memory cache. When it is full, we will apply our current list of items to process, and then flag the loop to run again.
+	maxMemoryTokens = 100000
```

Comment:
sorry, that was 1GB when using 1M and I set the value to 100,000 to be underneath that. I've dropped it again to 50,000.